pkg/apis/extensions/types.go

```diff
@@ -226,6 +226,14 @@ type DeploymentSpec struct {
 	// Value of this key is hash of DeploymentSpec.PodTemplateSpec.
 	// No label is added if this is set to empty string.
 	UniqueLabelKey string `json:"uniqueLabelKey,omitempty"`
+
+	// The config this deployment is rolling back to. Will be cleared after rollback is done.
+	Rollback *RollbackConfig `json:"rollback,omitempty"`
```

Comment:
Note that the reason that Binding contains ObjectMeta is so that annotations can be applied/updated. We should confirm that only the annotations field is used, but we should create a separate type for that purpose and file an issue to change Binding to use it also.

In this case, I think we may want to update the change-source annotation (and related annotations) on the Deployment. We could potentially automatically change it to mention the rollback, but I think it would be best to let kubectl decide the content of all the annotations it applies.